snapcraft/commands/remote.py

```diff
@@ -201,193 +137,21 @@ def _validate_single_artifact_per_build_on(self) -> None:
-                for arch, build_state in states.items():
-                    if build_state.is_running:
-                        building.add(arch)
-                    elif build_state == BuildState.SUCCESS:
-                        succeeded.add(arch)
-                    elif build_state == BuildState.UPLOADING:
-                        uploading.add(arch)
-                    elif build_state == BuildState.PENDING:
-                        pending.add(arch)
-                    else:
-                        not_building.add(arch)
-                progress_parts: list[str] = []
-                if not_building:
-                    progress_parts.append("Stopped: " + ",".join(sorted(not_building)))
-                if building:
-                    progress_parts.append("Building: " + ", ".join(sorted(building)))
-                if uploading:
-                    progress_parts.append("Uploading: " + ",".join(sorted(uploading)))
-                if succeeded:
-                    progress_parts.append("Succeeded: " + ", ".join(sorted(succeeded)))
-                if pending:
-                    progress_parts.append("Pending: " + ", ".join(sorted(pending)))
-                emit.progress("; ".join(progress_parts))
-        except TimeoutError:
-            if build_id:
-                resume_command = (
-                    f"{self._app.name} remote-build --recover --build-id={build_id}"
-                )
-            else:
-                resume_command = f"{self._app.name} remote-build --recover"
-            emit.message(
-                f"Timed out waiting for build.\nTo resume, run {resume_command!r}"
-            )
-            return 75  # Temporary failure
-
-        return_code = 0
-
-        for arch, build_state in states.items():
-            if build_state == BuildState.FAILED:
-                emit.progress(f"Build for architecture {arch} failed.", permanent=True)
-                return_code = 1
-
-        emit.progress(f"Fetching {len(builds)} build logs...")
-        logs = builder.fetch_logs(Path.cwd())
-        if not logs:
-            return_code = 1
-            emit.progress("No log files downloaded from Launchpad.", permanent=True)
-
-        emit.progress("Fetching build artifacts...")
-        artifacts = builder.fetch_artifacts(Path.cwd())
-        if not artifacts:
-            return_code = 1
-            emit.progress(
-                "No build artifacts downloaded from Launchpad.", permanent=True
-            )
-
-        log_names = sorted(path.name for path in logs.values() if path)
-        artifact_names = sorted(path.name for path in artifacts)
-
-        emit.message(
-            "Build completed.\n"
-            f"Log files: {', '.join(log_names)}\n"
-            f"Artifacts: {', '.join(artifact_names)}"
-        )
-        return return_code
-
-    def _get_archs(self, build_fors: list[str]) -> list[str]:
-        """Get the architectures to build for.
```

Comment:
WDYT of copying some or all of this docstring to `get_build_args()`?

Reply:
That's fine, but can you copy the raises, returns, and params parts of the docstring?